x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/PivotConfig.java

```diff
@@ -141,7 +149,67 @@ public boolean isValid() {
         return groups.isValid() && aggregationConfig.isValid();
     }
 
+    public List<String> aggFieldValidation() {
+        if ((aggregationConfig.isValid() && groups.isValid()) == false) {
+            return Collections.emptyList();
+        }
+        List<String> usedNames = new ArrayList<>();
+        // TODO this will need to change once we allow multi-bucket aggs + field merging
+        aggregationConfig.getAggregatorFactories().forEach(agg -> addAggNames(agg, usedNames));
+        aggregationConfig.getPipelineAggregatorFactories().forEach(agg -> addAggNames(agg, usedNames));
+        usedNames.addAll(groups.getGroups().keySet());
+        return aggFieldValidation(usedNames);
+    }
+
     public static PivotConfig fromXContent(final XContentParser parser, boolean lenient) throws IOException {
         return lenient ? LENIENT_PARSER.apply(parser, null) : STRICT_PARSER.apply(parser, null);
     }
+
+    /**
+     * Does the following checks:
+     *
+     *  - determines if there are any full duplicate names between the aggregation names and the group by names.
+     *  - finds if there are conflicting name paths that could cause a failure later when the config is started.
+     *
+     * Examples showing conflicting field name paths:
+     *
+     * aggName1: foo.bar.baz
+     * aggName2: foo.bar
+     *
+     * This should fail as aggName1 will cause foo.bar to be an object, causing a conflict with the use of foo.bar in aggName2.
+     * @param usedNames The aggregation and group_by names
+     * @return List of validation failure messages
+     */
+    static List<String> aggFieldValidation(List<String> usedNames) {
+        if (usedNames == null || usedNames.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<String> validationFailures = new ArrayList<>();
+
+        Set<String> leafNames = new HashSet<>(usedNames.size());
+        for(String name : usedNames) {
+            if (leafNames.contains(name)) {
+                validationFailures.add("duplicate field [" + name + "] detected");
+            }
+            leafNames.add(name);
+        }
+        usedNames.sort(String::compareTo);
+        for (int i = 0; i < usedNames.size() - 1; i++) {
```

Comment:
IIUC you can do the "duplicate field" validation in this loop as well (names are sorted so you can compare whether usedNames.get(i) is the same as usedNamed(i+1)) and get rid of lines 189-195.